src/file/mode.cr

```diff
@@ -0,0 +1,16 @@
+class ::File < IO::FileDescriptor
+  @[Flags]
+  enum Mode
+    Read
+    Write
+    Overwrite
```

Comment:
It's odd that this is not defined as `Overwrite = Write | Create | Truncate` and others.
I get that the meaning is given in `Crystal::System` but it due to the Flags, as long as the main values are difined everything should be covered.

Reply:
Right, I could totally do that!

I just wasn't thinking well when I wrote this commit last night :P